Contest_API.md

```diff
@@ -116,27 +116,36 @@ their expected behavior, if implemented.
     It can be used to request a whole collection or a specific element.
   - `POST`
     Create a new element. This can only be called on a collection
-    endpoint. No `id` attribute should be specified as it is up to the
-    server to assign one, which is returned in the location header.
+    endpoint. When using most [roles](#Roles) the `id` attribute may not
+    be specified as it is up to the server to assign one. When using an
+    `admin` role you may specify a new, unused id.
+    The response will contain a `Location` header pointing to the newly
+    created element and the response body will contain the initial
+    state of the element.
   - `PUT`
     Replaces a specific element. This method is idempotent and can only
     be called on a specific element and replaces its contents with the
     data provided. The payload data must be complete, i.e. no partial
     updates are allowed. The `id` attribute cannot be changed: it does
     not need to be specified (other than in the URL) and if specified
     different from in the URL, a `409 Conflict` HTTP code should be
-    returned.
+    returned. The response body will contain the updated state of the
+    element.
   - `PATCH`
     Updates/modifies a specific element. Similar to `PUT` but allows
     partial updates by providing only that data, for example:
     `PATCH  https://example.com/api/contests/wf14/teams/10`
     with JSON contents
-    `{"name":"Our cool new team name"}`
-    No updates of the `id` attribute are allowed either.
+    `{"name":"Our cool new team name"}`.
+    The `id` attribute cannot be changed: it does
+    not need to be specified (other than in the URL) and if specified
```

Comment:
Same as above: I think `id`s should be required.

Reply:
I think for `PATCH` this is not true.

Reply:
Agreed with Nicky, PATCH should only be required to include attributes you want to change.

Reply:
Yeah, you're absolutely right.